compiler/src/dotty/tools/dotc/Driver.scala

```diff
@@ -30,18 +30,20 @@ class Driver {
 
   protected def doCompile(compiler: Compiler, files: List[AbstractFile])(using Context): Reporter =
     if files.nonEmpty then
+      var runOrNull = ctx.run
       try
         val run = compiler.newRun
+        runOrNull = run
         run.compile(files)
         finish(compiler, run)
       catch
         case ex: FatalError =>
           report.error(ex.getMessage.nn) // signals that we should fail compilation.
-        case ex: TypeError =>
-          println(s"${ex.toMessage} while compiling ${files.map(_.path).mkString(", ")}")
+        case ex: TypeError if !runOrNull.enrichedErrorMessage =>
```

Comment:
Wondering: Why can we call `.enrichErrorMessage` on runOrNull? Does that not violate null-safety?

Oh I see now, it's because there is an extension method on `Run | Null`! I think that's cute but maybe a bit surprising? One can avoid the surprise by a comment here.